torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -2487,6 +2488,78 @@ void fuseAllLoops(StmtPtr st) {
   }
 }
 
+// Prune innermost loops until iterations satisfies a minimum grain size.
+static void pruneByGrainSize(std::vector<ForPtr>& loops) {
+  constexpr int64_t minGrainSize = 32768;
+  int64_t grainSize = 1;
+  for (int64_t i = loops.size(); i > 0; i--) {
+    auto const& loop = loops[i - 1];
+    if (auto stop = to<IntImm>(loop->stop())) {
+      grainSize *= stop->value();
```

Comment:
This is assuming the loops are normalized at this point. While that might be true mostly, I'm not sure that is guaranteed.

Reply:
I don't think anything's likely to un-normalize the loops before this point, but maybe I should simplify (and do stop-start) just to be safe.  I guess the worst that happens is we miss a parallelization opportunity.